src/main/java/seedu/address/logic/commands/ArchiveCommand.java

```diff
@@ -0,0 +1,198 @@
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Person;
+import seedu.address.model.util.DeliveryAction;
+import seedu.address.ui.InspectWindow;
+
+
+/**
+ * Archive a delivery identified using it's displayed index from the address book.
+ */
+
+public class ArchiveCommand extends Command {
+
+    public static final String COMMAND_WORD = "archive";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Archives the delivery identified by the index number used in the displayed delivery list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_ARCHIVED_DELIVERY_SUCCESS = "Archived Delivery for %1$s: %2$s";
+
+    private final List<Index> indexList;
+
+    public ArchiveCommand(List<Index> indexList) {
+        this.indexList = indexList;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        // Sort the indexList in descending order
+        indexList.sort(Comparator.comparing(Index::getZeroBased).reversed());
+
+        if (!AddressBookParser.getInspect()) {
+            return handleDeliveryArchive(model);
+        } else {
+            throw new CommandException("Cannot archive contact!");
+        }
+    }
+
+    /**
+     * Handles the deletion of deliveries from the inspected person's delivery list based on the indexList.
+     *
+     * @param model The model containing the inspected person.
+     * @return A CommandResult containing a success message with details of the deleted deliveries.
+     * @throws CommandException if any index in the indexList is out of bounds or if duplicates are found.
+     */
+    private CommandResult handleDeliveryArchive(Model model) throws CommandException {
+        requireNonNull(model);
+        Person inspectedPerson = InspectWindow.getInspectedPerson();
+        List<Delivery> deliveryList = inspectedPerson.getDeliveryList();
+        validateIndexes(inspectedPerson.getDeliveryListSize(), indexList);
+
+        List<Delivery> deliveryToArchiveList = archiveDeliveries(inspectedPerson, deliveryList);
+
+        return new CommandResult(String.format(
+                MESSAGE_ARCHIVED_DELIVERY_SUCCESS,
+                inspectedPerson.getName(),
+                Messages.formatDeliveryList(reverseList(deliveryToArchiveList))),
+                DeliveryAction.ARCHIVE);
+    }
+
+    /**
+     * Validates the indexes in the indexList to ensure none are out of bounds or duplicates.
+     *
+     * @param listSize The size of the list from which items are to be deleted.
+     * @param indexList The list of indexes to be validated.
+     * @throws CommandException if any index is out of bounds or if duplicates are found.
+     */
+    private void validateIndexes(int listSize, List<Index> indexList) throws CommandException {
+        boolean duplicate = hasDuplicates(indexList);
```

Comment:
Should this be changed to isDuplicate, hasDuplicate, or something similar to adhere more closely to the coding standard?

Reply:
Can I clarify for the change you referring to, is it to the method name `validateIndexes` or the method `hasDuplicates` used?

Reply:
I was only referring to the variable `duplicate`, the rest seems good 👍